schemas/event.yml

```diff
@@ -38,53 +38,174 @@
       example: 4648
 
     - name: kind
-      level: extended
+      level: core
       type: keyword
-      short: The kind of the event.
+      short: The kind of the event. (Categorization Field)
       description: >
-        The kind of the event.
-
-        This gives information about what type of information the event
-        contains, without being specific to the contents of the event.
-      example: event
-      accepted_values:
-      - name: alert
-        description: >
-          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
-          incididunt ut labore et dolore magna aliqua.
-      - name: event
-        description: >
-          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
-          incididunt ut labore et dolore magna aliqua.
+        This is one of four ECS Categorization Fields, and indicates the
+        highest level in the ECS category hierarchy.
+
+        `event.kind` gives high-level information about what type of information the
+        event contains, without being specific to the contents of the event.
+        For example, values of this field distinguish alert events from
+        metric events.
+      example: alert
+      allowed_values:
+        - name: alert
+          description: >
+            `event.kind:alert` indicates an event that describes an alert.
+            Alerts are often associated with detection rules. `event.kind:alert`
+            is often populated for events coming from firewalls, intrusion detection
+            systems, endpoint detection and response systems, etc.
+            It is used to indicate that an alert was triggered.
+        - name: event
+          description: >
+            `event.kind:event` is the most general and most common value of this
+            field. It is used to represent events that indicate that something happened.
+        - name: metric
+          description: >
+            Used to indicate that this event is a measurement taken at given point in time.
+            Examples include CPU utilization, memory usage, or a vulnerability scan result.
+            Events with `event.kind:metric` indicate that a measurement was taken.
+        - name: state
+          description: >
+            This value is similar to metric, except that the entity being measured does not
+            provide a numeric metric value, but rather one of a fixed set of conditions or states.
+            For example a periodic event reporting a "fin_wait" state of a TCP connection
+            on a host might use `event.type:state`.
```

Comment:
I wonder if we should hold off on releasing this one until we've confirmed the semantics a little more. Here are two reservations I still have, at this time:

The state that can be observed at one moment in time (the cluster is green). The state transitions (e.g. cluster goes from yellow to green). This definition should address whether both kinds of events belong in event.kind:state. My understanding is that only "point in time" observations, and not the state transitions should belong here. I'm not sure if we have agreement on that yet.

Another worry I have is that in many cases, checking on the state of a system can actually report back metrics. So in these cases, these state checks would be `event.kind:metric` because they contain metrics. If that's the case, how useful/reliable will `event.kind:state` be? Perhaps it's better to have all state checks in event kind metric?

Reply:
If there's not clarity in what this category is used for, I'd tend to agree. **But** isn't there agreement on what this field is used for? If the answer is "no" we should strip it out, but if others have been thinking of this the same way I don't see a need to remove it from the initial release.

As you state, the language seems to indicate any sort of data state falls under data involving "one of a fixed set of conditions or states." I'm not sure that there's really a meaningful distinction between "heartbeat" style periodic events or events that have the same exact data, but are only fired on state transitions--so I would imagine that this would cover either use-case.

Just for illustration purposes, what I've been thinking of using this category for is for any sort of Endpoint internal state transitions--i.e. policy applications, upgrades, lifecycle, etc.--so besides the surrounding metadata of things like versions, categorization fields, and host-level information, I envision that implementations that would use this category similarly to how I envisioned would contain a lot of hybrid data, with much "application-specific state" in custom fields. That's how I see this differ from the idea of "metrics."

Reply:
Now that we've agreed on adding a beta warning to the section, I'm less worried here.

Let's leave it in for now and see if we take it out later.